src/store/messagesStore.js

```diff
@@ -400,7 +400,16 @@ const actions = {
 		if (updateVisually) {
 			context.commit('setVisualLastReadMessageId', { token, id })
 		}
-		await updateLastReadMessage(token, id)
+
+		if (context.getters.getUserId()) {
+			// only update on server side if there's an actual user, not guest
+			await updateLastReadMessage(token, id)
+		}
+
+		if (updateVisually) {
+			// refetch to get more accurate values for unreadMessages, unreadMention, etc
+			context.dispatch('fetchConversation', { token })
```

Comment:
this will bring down our server in the company call
If someone writes a message everyone will mostly simultaniously pull the room endpoint to update, or when is `updateVisually` set to true?

Reply:
`updateVisually` in this code path is only true for direct forcing-actions like here "mark as unread".
the actual visual update is happening in a separate action

but you're right to be careful, I'll double check to make sure it's not called more than that

Reply:
I've checked and while it wouldn't be called too often, it still would be called when posting a new message because that's the moment where we update the indicator visually also.

I've now moved the `fetchConversation` to only happen when using "mark as unread", as it's the only situation where we'd have non-zero `unreadMessages` after an action.